cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/input/DelegatingRecordReader.java

```diff
@@ -70,7 +74,12 @@ public void initialize(InputSplit split, TaskAttemptContext context) throws IOEx
     // We need to be sure not to pass the TaggedInputSplit to the underlying RecordReader. Otherwise, it can result
     // in ClassCastExceptions
     InputSplit inputSplit = ((TaggedInputSplit) split).getInputSplit();
-    originalRR.initialize(inputSplit, context);
+    Job jobCopy = new Job(context.getConfiguration());
```

Comment:
You don't need to create a Job to copy configuration. You just need to do `new Configuration(context.getConfiguration());`